RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  BaseManageSubscriptionViewModel.swift
+//
+//  Created by Facundo Menzella on 5/5/25.
+
+import Foundation
+@_spi(Internal) import RevenueCat
+import SwiftUI
+
+#if os(iOS)
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@MainActor
+class BaseManageSubscriptionViewModel: ObservableObject {
```

Comment:
I usually try to avoid subclassing, but here it felt like an easy workaround to reuse path logic. It's good enough for now I think